src/game/common/thing/thingtemplate.h

```diff
@@ -158,21 +158,25 @@ class ThingTemplate : public Overridable
     static void Parse_Max_Simultaneous(INI *ini, void *instance, void *store, const void *user_data);
     static void Parse_Module_Name(INI *ini, void *instance, void *store, const void *user_data);
     static void Parse_Per_Unit_FX(INI *ini, void *instance, void *store, const void *user_data);
+    static void Parse_Arbitrary_FX_Into_Map(INI *ini, void *instance, void *store, const void *user_data);
     static void Parse_Per_Unit_Sounds(INI *ini, void *instance, void *store, const void *user_data);
+    static void Parse_Arbitrary_Sounds_Into_Map(INI *ini, void *instance, void *store, const void *user_data);
     static void Parse_Prerequisites(INI *ini, void *instance, void *store, const void *user_data);
+    static void Parse_Prerequisite_Unit(INI *ini, void *instance, void *store, const void *user_data);
+    static void Parse_Prerequisite_Science(INI *ini, void *instance, void *store, const void *user_data);
     static void Parse_Remove_Module(INI *ini, void *instance, void *store, const void *user_data);
     static void Parse_Replace_Module(INI *ini, void *instance, void *store, const void *user_data);
     static void Parse_Weapon_Template_Set(INI *ini, void *instance, void *store, const void *user_data);
     static void Parse_Overrideable_By_Like_Kind(INI *ini, void *instance, void *store, const void *user_data);
 
-    int Calc_Cost_To_Build(const Player *player) const;
-    int Calc_Time_To_Build(const Player *player) const;
+    int Calc_Cost_To_Build(const Player *player) const; // needs more of Player
+    int Calc_Time_To_Build(const Player *player) const; // needs more of Player
     bool Can_Possibly_Have_Any_Weapon() const;
     void Copy_From(const ThingTemplate *that);
     const ArmorTemplateSet *Find_Armor_Template_Set(const BitFlags<ARMORSET_COUNT> &t) const;
     const WeaponTemplateSet *Find_Weapon_Template_Set(const BitFlags<WEAPONSET_COUNT> &t) const;
     AIUpdateModuleData *Friend_Get_AI_Module_Info() const;
-    ThingTemplate *Get_Build_Facility_Template(const Player *player) const;
+    ThingTemplate *Get_Build_Facility_Template(const Player *player) const; // needs more of Player
```

Comment:
you mean ProductionPrerequisite?

Reply:
No, it needs more of Player in order to be able to implement the bits of ProductionPrerequisite bits needed for this function.